blockchain/fullblocktests/generate.go

```diff
@@ -1462,17 +1554,75 @@ func Generate(includeLargeReorg bool) (tests [][]TestInstance, err error) {
 	//                 \-> b51(15)
 	g.SetTip("b46")
 	g.NextBlock("b51", outs[15], ticketOuts[15], func(b *wire.MsgBlock) {
+		// set the merkle root to an invalid hash
 		b.Header.MerkleRoot = chainhash.Hash{}
 	})
 	g.AssertTipBlockMerkleRoot(chainhash.Hash{})
 	rejected(blockchain.ErrBadMerkleRoot)
 
-	// Create block with an invalid proof-of-work limit.
+	// Create block with an invalid stake root
 	//
 	//   ... -> b46(14)
 	//                 \-> b52(15)
 	g.SetTip("b46")
 	g.NextBlock("b52", outs[15], ticketOuts[15], func(b *wire.MsgBlock) {
+		// set the stake root to an invalid hash
+		b.Header.StakeRoot = chainhash.Hash{}
+	})
+	g.AssertTipBlockStakeRoot(chainhash.Hash{})
+	rejected(blockchain.ErrBadMerkleRoot)
+
+	// Create block with an invalid block size
+	//
+	//   ... -> b46(14)
+	//                 \-> b53(15)
+	g.SetTip("b46")
+	g.NextBlock("b53", outs[15], ticketOuts[15], func(b *wire.MsgBlock) {
+		// set the block size an invalid value
+		b.Header.Size = 0x20ffff71
+	})
+	rejected(blockchain.ErrWrongBlockSize)
+
+	// Create block with an invalid subsidy for a coinbase input.
+	//
+	//   ... -> b46(14)
+	//                 \-> b54(15)
+	g.SetTip("b46")
+	g.NextBlock("b54", outs[15], ticketOuts[15], func(b *wire.MsgBlock) {
+		// set the coinbase subsidy to invalid value (greater than coin supply)
+		b.Transactions[0].TxIn[0].ValueIn = 1234567890123456
```

Comment:
Rather than setting a specific number, I would suggest doing `b.STransactions[0].TxIn[0].ValueIn++` so that the test can't possibly create any false positives if the actual value in somehow ultimately ends up being the magic number.